std/algorithm/iteration.d

```diff
@@ -1883,6 +1887,19 @@ if (isForwardRange!Range)
     assert(grp1.save.equal([1, 3, 5]));
 }
 
+// Issue 18751
+unittest
```

Comment:
Ouch.  I added `@safe` and suddenly dmd is complaining:
````
std/algorithm/iteration.d(1899): Error: @safe function std.algorithm.iteration.__unittest_L1891_C7 cannot call @system function std.algorithm.iteration.__unittest_L1891_C7.chunkBy!((i, j) => data[i] == data[j], int[]).chunkBy
std/algorithm/iteration.d(1899): Error: @safe function std.algorithm.iteration.__unittest_L1891_C7 cannot call @system destructor std.algorithm.iteration.__unittest_L1891_C7.ChunkByImpl!(__lambda1, int[]).ChunkByImpl.~this
std/algorithm/iteration.d(1900): Error: @safe function std.algorithm.iteration.__unittest_L1891_C7 cannot call @system function std.algorithm.comparison.equal!(equal).equal!(ChunkByImpl!(__lambda1, int[]), int[][]).equal
std/algorithm/iteration.d(1900): Error: @safe function std.algorithm.iteration.__unittest_L1891_C7 cannot call @system generated function std.algorithm.iteration.__unittest_L1891_C7.ChunkByImpl!(__lambda1, int[]).ChunkByImpl.__fieldPostblit
````

I've no idea what happened; I literally just copy-n-pasted the implementation from the original code pretty much untouched, except to add template arguments so that it will compile.

Reply:
Any ideas?

Reply:
Sorry. I forgot that chunkBy isn't @safe.
Just @system then :/

Reply:
(this is due to its use of `RefCounted` under the hood which isn't @safe)